src/main/java/by/yahor_kulesh/mappers/TicketMapper.java

```diff
@@ -0,0 +1,105 @@
+package by.yahor_kulesh.mappers;
+
+import by.yahor_kulesh.entity.TicketEntity;
+import by.yahor_kulesh.entity.UserEntity;
+import by.yahor_kulesh.entity.enums.TicketType;
+import by.yahor_kulesh.model.tickets.BusTicket;
+import by.yahor_kulesh.model.tickets.ConcertTicket;
+import by.yahor_kulesh.model.tickets.Ticket;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.Mappings;
+import org.mapstruct.Named;
+import org.mapstruct.ObjectFactory;
+import org.mapstruct.factory.Mappers;
+
+import java.util.Set;
+import java.util.UUID;
+
+
+@Mapper
+public interface TicketMapper extends CommonMapper{
+
+    TicketMapper INSTANCE = Mappers.getMapper(TicketMapper.class);
+
+
+    @ObjectFactory
+    default Ticket toModel(TicketEntity ticketEntity){
+        if(ticketEntity == null) return null;
+        if(TicketType.BUS == ticketEntity.getType()) {
+            return toBus(ticketEntity);
+        } else if(ticketEntity.getType()==TicketType.CONCERT) {
+            return toConcert(ticketEntity);
+        } else return toTicket(ticketEntity);
+    }
+
+    @Mappings({
+            @Mapping(target = "creationTime", source = "creationTime", qualifiedByName = "timestampToZonedDateTime"),
+            @Mapping(target = "userId", source="user", qualifiedByName = "userEntityToUUID")
+    })
+    ConcertTicket toConcert(TicketEntity ticketEntity);
```

Comment:
the same with the name of methods:
`ConcertTicket toConcertTicket(TicketEntity ticketEntity);`
instead of
`ConcertTicket toConcert(TicketEntity ticketEntity);`